lib/GIL/Module.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Module.hpp"
+
+namespace glu::gil {
+
+void Module::addFunction(
+    std::string name, glu::types::FunctionTy *type,
+    std::list<BasicBlock> basicBlocks
+)
+{
+    Function *f = new Function(name, type);
+
+    for (auto &bb : basicBlocks)
+        f->addBasicBlockAtEnd(&bb);
+    _functions.push_back(std::move(f));
+};
+
+Function const *Module::getFunction(std::string name) const
+{
+    for (auto &f : _functions)
+        if (f.getName() == name)
+            return &f;
+    return nullptr;
+};
+
+void Module::deleteFunction(Function *f)
+{
+    _functions.remove(f);
+};
```

Comment:
delete f?

Reply:
when you call remove() it calls delete_node() which already deletes the function f